groovy-language-server/src/main/java/com/palantir/ls/groovy/util/SourceWriter.java

```diff
@@ -69,6 +69,19 @@ public Path getDestination() {
      * Applies the given changes to the destination file. Does not handle intersecting ranges in the changes.
      */
     public synchronized void applyChanges(List<TextDocumentContentChangeEvent> contentChanges) throws IOException {
+        // Check if any of the ranges are null
+        for (TextDocumentContentChangeEvent change : contentChanges) {
+            if (change.getRange() == null) {
+                checkArgument(contentChanges.size() == 1,
+                        String.format(
+                                "Cannot have many changes when a change contains a null range which means it replaces "
```

Comment:
something like `Cannot handle more than one change in one single request when some of them have a null range`? Probably don't need to explain what null range means